src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -519,6 +521,51 @@ public <T> void getProvisioningProgress(
         }
     }
 
+    /**
+     * Check workflow provisioning state and resources to see if state can be deleted with template
+     *
+     * @param documentId document id
+     * @param canDeleteStateConsumer consumer function which will be true if NOT_STARTED or COMPLETED and no resources
+     * @param listener action listener from caller to fail on error
+     * @param <T> action listener response type
+     */
+    public <T> void canDeleteWorkflowStateDoc(String documentId, Consumer<Boolean> canDeleteStateConsumer, ActionListener<T> listener) {
```

Comment:
Since it's boolean we can use `Predicate` here instead of a `Consumer`. More suitable for checking conditions.

Reply:
I copied what was done elsewhere, please submit a new refactor issue with this proposal and tag it "good first issue".